app/models/manageiq/providers/openstack/infra_manager/host/operations.rb

```diff
@@ -0,0 +1,40 @@
+module ManageIQ::Providers::Openstack::InfraManager::Host::Operations
+  def self.included(base)
+    base.send :include, InstanceMethods
+    base.extend ClassMethods
+  end
+
+  module InstanceMethods
+    def nova_system_service
+      # we need to be sure that host has compute service
+      system_services.find_by(:name => 'openstack-nova-compute')
+    end
+
+    def nova_fog_service
+      # TODO: check if host is part of OpenStack Infra
+      # host's cluster needs cloud assigned
+      cloud = ems_cluster.cloud
+      # hostname of host in hypervisor is used to properly select service from OpenStack
+      host_name = hypervisor_hostname
+      fog_services = cloud.openstack_handle.compute_service.services
+      fog_services.find { |s| s.host =~ /#{host_name}/ && s.binary == 'nova-compute' }
+    end
+
+    def nova_fog_enable_service
+      nova_fog_service.enable
+    end
+
+    def nova_fog_disable_service
```

Comment:
There was a comment by @blomquisg on my PR suggesting the removal of the fog-related part in the method name -- i.e. ironic_fog_set_node_maintenance shortened to set_node_maintenance. I'm not sure what the convention is for provider API calls that call out to fog, but these method names would be in a similar category -- whatever we do should probably be consistent across these features. Here's a link to the comment:
https://github.com/ManageIQ/manageiq/pull/7528#discussion_r58626299